### ISSUE TITLE
paste_linter should not warn inside expression() in a plot

### ISSUE THREAD

**Issue by mcol** [r-lib/lintr#2945]
This suggestion is actually misleading, as only `paste()` and not `paste0()` is allowed when `expression()` is used within a plot, such as to generate title or axis labels:
```R
warning: [paste_linter] paste0(...) is better than paste(..., sep = "").
      main = expression(paste("Plateau test ", L[n], ",", L[x], " curves", sep =
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

The rationale is that only a certain number of functions and operators are valid inside `expression()`:
https://stat.ethz.ch/R-manual/R-devel/library/grDevices/html/plotmath.html
